### PR TITLE
Remove block dependency from block/utils - Closes #4012

### LINKED ISSUE #4012: Remove block dependency from blocks/utils
### Expected behavior
Blocks/Utils must be pure utility functions without have any dependency on any other module. 

### Actual behavior

Currently the `block/utils.js` is requiring `block.js` which is requiring the `block_v1.js` and `block_v2.js`. It violates the concepts of utility functions. Also in case we need to use some utility function inside `block_v1.js` or `block_v2.js`, it creates a circular dependency and failed. 

https://github.com/LiskHQ/lisk-sdk/blob/8920e9dd4bd1be8aa578d294395fe46d50b0f970/framework/src/modules/chain/blocks/utils.js#L20

### Which version(s) does this affect? (Environment, OS, etc...)
2.3.0

### PR DESCRIPTION
**Pull request by michielmulders**
### What was the problem?
Passing block dependency to several files in the `block` folder creating circular dependency.
To solve this now, we pass the `blocksLogic` dependency.

However, as @nazarhussain indicated, we should create pure functions as util functions. Currently, they do processing and `I/O` with DB. 

### How did I solve it?
Pass `blocksLogic` dependency directly as argument.

### Review checklist

- [x] The PR resolves #4012
- [x] All new code is covered with unit tests
- [x] Relevant integration / functional tests are added
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [x] Documentation has been added/updated

### CODE REVIEW

**Comment by michielmulders** [comment]
To avoid circular dependency @shuse2

**Comment by shuse2** [comment]
I would prefer to clean up the code location than just injecting for that purpose

File: framework/src/modules/chain/blocks/blocks.js [hunk 1]
```diff
@@ -293,6 +294,7 @@ class Blocks extends EventEmitter {
 				this.storage,
 				this.interfaceAdapters,
 				this.genesisBlock,
+				blocksLogic,
```
Comment (nazarhussain) [inline framework/src/modules/chain/blocks/blocks.js]: Agree that its the easiest way to solve. But I don't like that direction. We have to resolve and not pass any other module to utils. To me when util functions require instance of other modules then these are actually not standalone utils functions. We should rather move `loadLastBlock` to `blockLogic` where it relates more accurately.

Comment (michielmulders) [reply framework/src/modules/chain/blocks/blocks.js]: The loadLastBlock function is not the issue, it is that `dbRead` and `storageRead` require `blocksLogic`.

**Comment by michielmulders** [comment]
@nazarhussain I moved the functions that were not pure util functions to `block.js`.
Also, implemented back the `sortTransactions` function which was initially causing the circular dep.

Currently, the dep tree looks like:
`block` -> `block_v1/2` -> `utils`

`blocks`
    -> `utils`
    -> `block`

`utils` -> No dependencies!

**Full dep list: `blocks` -> `block` -> `block_v1/2` -> `utils` END**

Before:
`utils` required `block` which caused the circular dependency.

File: framework/src/modules/chain/blocks/blocks.js [hunk 2]
```diff
@@ -17,6 +17,7 @@
 const EventEmitter = require('events');
 const { cloneDeep } = require('lodash');
 const blocksUtils = require('./utils');
+const Block = require('./block');
```
Comment (shuse2) [inline framework/src/modules/chain/blocks/blocks.js]:
```suggestion
const blocksLogic = require('./block');
```

**Pull request merged**